src/storage/src/sst.rs

```diff
@@ -285,13 +295,53 @@ impl FileHandleInner {
     }
 }
 
+#[derive(Debug, Snafu)]
+pub struct ParseIdError {
+    source: uuid::Error,
+}
+
+/// Unique id for [SST File].
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, Serialize, Deserialize)]
+pub struct FileId(Uuid);
+
+impl FileId {
+    /// Returns a new unique [FileId] randomly.
+    pub fn random() -> FileId {
+        FileId(Uuid::new_v4())
+    }
+
+    /// Parses id from string.
+    pub fn parse_str(input: &str) -> std::result::Result<FileId, ParseIdError> {
+        Uuid::parse_str(input).map(FileId).context(ParseIdSnafu)
+    }
+
+    /// Append `.parquet` to file id to make a complete file name
+    pub fn as_parquet(&self) -> String {
+        format!("{}{}", self.0.hyphenated(), ".parquet")
+    }
+}
+
+impl fmt::Display for FileId {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}", self.0)
+    }
+}
+
+impl FromStr for FileId {
+    type Err = ParseIdError;
+
+    fn from_str(s: &str) -> std::result::Result<FileId, ParseIdError> {
+        FileId::parse_str(s)
+    }
+}
+
 /// Immutable metadata of a sst file.
 #[derive(Clone, Debug, PartialEq, Eq, Hash, Serialize, Deserialize)]
 pub struct FileMeta {
     /// Region of file.
     pub region_id: RegionId,
-    /// File name
-    pub file_name: String,
+    /// Compared to normal file names, FileId ignore the extension
+    pub file_id: FileId,
```

Comment:
`FileMeta` is written to region manifest. changed `file_name` to `file_id` will cause panic when parsing existing manifest files.

Reply:
I see. So, can we use a reading/parsing funtion that is able to read either `file_id` or `file_name`? Then the history files and future files can be compatible.

Reply:
> I see. So, can we use a reading/parsing funtion that is able to read either `file_id` or `file_name`? Then the history files and future files can be compatible. Or when writing to manifest files, append file extension to make them in the same format with existed manitest files.

I came up with an idea:

Use the old field name `file_name` instead of `file_id` to ensure old field can be recognized, and use customized deserialization function to deserialize existing manifest with ".parquet" suffix.
```rs
pub struct FileMeta {
    pub region_id: RegionId,
    #[serde(deserialize_with = "deserialize_from_string")]
    pub file_name: FileId,
    //...
}

fn deserialize_from_string<'de, D>(deserializer: D) -> std::result::Result<FileId, D::Error>
where
    D: Deserializer<'de>,
{
    let s: &str = Deserialize::deserialize(deserializer)?;
    let stripped = s.strip_suffix(".parquet").unwrap_or(s); // strip parquet suffix if needed.
    FileId::from_str(stripped).map_err(|e| D::Error::custom(e))
}
```

Reply:
Looks good. With this function, we can extract xxx from "xxx.parquet" as well as xxx from "xxx". And in the following flushed `FileMeta` can be without ".parquet".
The only problem is `file_name` field contains a value of type `FileId`. It's a little weird. So maybe I'll rename the type as `FileName`? We can discuss it.

Reply:
done!